packages/shell-api/src/helpers.ts

```diff
@@ -644,10 +668,83 @@ export async function getPrintableShardStatus(
     )
   ).filter((dbEntry) => !!dbEntry);
 
+  let shardedDataDistribution: ShardedDataDistribution | undefined;
+
+  try {
+    // Available since 6.0.3
+    const adminDb = db.getSiblingDB('admin');
+    shardedDataDistribution = (await (
+      await adminDb.aggregate([{ $shardedDataDistribution: {} }])
+    ).toArray()) as ShardedDataDistribution;
+  } catch (e) {
+    // Pass, most likely an older version.
+    console.error('shardedDataDistribution Errored:', e);
+  }
+
+  result.shardedDataDistribution = shardedDataDistribution;
+
   delete result.shardingVersion.currentVersion;
   return result;
 }
 
+type ShardingStatusResult = {
+  shardingVersion: {
+    _id: number;
+    clusterId: ObjectId;
+    /** This gets deleted when it is returned from getPrintableShardStatus */
+    currentVersion?: number;
+  };
+  shards: {
+    _id: string;
+    host: string;
+    state: number;
+    topologyTime: Timestamp;
+    replSetConfigVersion: Long;
+  }[];
+  [mongoses: `${string} mongoses`]:
+    | 'none'
+    | {
+        [version: string]:
+          | number
+          | {
+              up: number;
+              waiting: boolean;
+            };
+      }[];
+  autosplit: {
+    'Currently enabled': 'yes' | 'no';
+  };
+  balancer: {
+    'Currently enabled': 'yes' | 'no';
+    'Currently running': 'yes' | 'no' | 'unknown';
+    'Failed balancer rounds in last 5 attempts': number;
+    'Migration Results for the last 24 hours':
+      | 'No recent migrations'
+      | {
+          [count: number]:
+            | 'Success'
+            | `Failed with error '${string}', from ${string} to ${string}`;
+        };
+    'Balancer active window is set between'?: `${string} and ${string} server local time`;
+    'Last reported error'?: string;
+    'Time of Reported error'?: string;
+    'Collections with active migrations'?: `${string} started at ${string}`[];
+  };
+  shardedDataDistribution?: ShardedDataDistribution;
+  databases: { database: Document; collections: Document }[];
+};
+
+export type ShardedDataDistribution = {
```

Comment:
Don't know where to leave this comment, but I noticed on the ticket the comment:  "if it's possible I think the order should be ownedSizeBytes, numOwnedDocuments, orphanedSizeBytes, numOrphanedDocs".

Wondering how much effort we should put in for that request. Probably not worth it and kinda weird precedent to be setting. Wouldn't want sorting/ordering to break this when they add fields, for example.

Reply:
Yeah, wanted to comment on that too; I think the idea of controlling ordering is weird, I imagine it'd be consistent with whatever order the DB returns and I doubt that'd change much if at all. I will try to run it on an actual and see if that's the case.